include/bluetooth/audio/audio.h

```diff
@@ -1,9 +1,1690 @@
+	 *  released and can be deallocated.
+	 *
+	 *  @param stream Stream object that has been released.
+	 */
+	void (*released)(struct bt_audio_stream *stream);
+
+	/** @brief Stream connected callback
+	 *
+	 *  If this callback is provided it will be called when the
+	 *  isochronous stream is connected.
+	 *
+	 *  @param stream The stream that has been connected
+	 */
+	void (*connected)(struct bt_audio_stream *stream);
+
+	/** @brief Stream disconnected callback
+	 *
+	 *  If this callback is provided it will be called when the
+	 *  isochronous stream is disconnected, including when a connection gets
+	 *  rejected.
+	 *
+	 *  @param stream The stream that has been Disconnected
+	 *  @param reason HCI reason for the disconnection.
+	 */
+	void (*disconnected)(struct bt_audio_stream *stream, uint8_t reason);
+
+	/** @brief Stream audio HCI receive callback.
+	 *
+	 *  This callback is only used if the ISO data path is HCI.
+	 *
+	 *  @param stream Stream object.
+	 *  @param buf  Buffer containing incoming audio data.
+	 */
+	void (*recv)(struct bt_audio_stream *stream, struct net_buf *buf);
+};
+
+/** @brief Audio Capability type */
+enum bt_audio_pac_type {
+	BT_AUDIO_SINK = 0x01,
+	BT_AUDIO_SOURCE = 0x02,
+} __packed;
+
+/** @brief Register Audio callbacks for a stream.
+ *
+ *  Register Audio callbacks for a stream.
+ *
+ *  @param stream Stream object.
+ *  @param ops    Stream operations structure.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+void bt_audio_stream_cb_register(struct bt_audio_stream *stream,
+				 struct bt_audio_stream_ops *ops);
+
+/** @brief Register unicast server callbacks.
+ *
+ *  Only one callback structure can be registered, and attempting to
+ *  registering more than one will result in an error.
+ *
+ *  @param cb  Unicast server callback structure.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_audio_unicast_server_register_cb(const struct bt_audio_unicast_server_cb *cb);
+
+/** @brief Unregister unicast server callbacks.
+ *
+ *  May only unregister a callback structure that has previously been
+ *  registered by bt_audio_unicast_server_register_cb().
+ *
+ *  @param cb  Unicast server callback structure.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_audio_unicast_server_unregister_cb(const struct bt_audio_unicast_server_cb *cb);
+
+/**
+ * @defgroup bt_audio_client Audio Client APIs
+ * @ingroup bt_audio
+ * @{
+ */
+
+struct bt_audio_discover_params;
+
+/** @typedef bt_audio_discover_func_t
+ *  @brief Discover Audio capabilities and endpoints callback function.
+ *
+ *  If discovery procedure has complete both cap and ep are set to NULL.
+ */
+typedef void (*bt_audio_discover_func_t)(struct bt_conn *conn,
+					 struct bt_codec *codec,
+					 struct bt_audio_ep *ep,
+					 struct bt_audio_discover_params *params);
+
+struct bt_audio_discover_params {
+	/** Capabilities type */
+	uint8_t  type;
```

Comment:
Here's one example of ignoring the enum type itself and just going with an `uint8_t`. If you really want to use a packed enum, take a look at `bt_conn_state_t`. Otherwise, just remove the `__packed` from the definition since it doesn't give you any benefit.